src/test/common/clara/test_simple_rules.cljc

```diff
@@ -194,3 +194,14 @@
     (is (has-fact? (:cold @side-effect-holder) (->Temperature -10 "MCI") ))
 
     (is (has-fact? (:subzero @side-effect-holder) (->Temperature -10 "MCI")))))
+
+(def-rules-test test-condition-with-empty-constraint
+  {:rules [simple-rule [[[Cold ()]]
```

Comment:
Can we put the empty list somewhere also where serialization of it will be required even if the empty constraint is filtered out?  Say [Cold () (not= temperature ()]

Reply:
Added this to the basic and deserialization test